### PR TITLE
CED-1444: horizontal list

### PR DESCRIPTION
**Pull request by nathanielwarner**
### 🔗 Linked issue

https://energysage.atlassian.net/browse/CED-1444

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Updated `EsHorizontalList` to the ESDS 2.0 theme

### 🥼 Testing

manual testing

#### 🧐 Feedback Requested / Focus Areas

- There's a discrepancy in the figma, as it says "Disabled icon is grey 500", but it's actually displaying gray 600. I'm using gray 500 since that's explicitly stated, and matches what's used in the pagination component.
- This component is custom-designed, and is not accessible at all. I suppose it's outside the scope of this PR to fix, but what do you think about adding a note to the documentation of this component that using it should be avoided due to accessibility? In our repos, I see it's used for `ArticleRelated` currently.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.
- [x] I have documented testing approach

### CODE REVIEW

File: es-vue-base/src/lib-components/EsHorizontalList.vue [hunk 1]
```diff
@@ -28,7 +28,11 @@
             <es-button
                 v-if="arrows"
                 variant="link"
-                class="text-black d-none d-lg-block p-0 mr-200"
+                :disabled="atBeginning"
+                class="d-none d-lg-block p-0 mr-200"
```
Comment (ericdouglaspratt) [inline es-vue-base/src/lib-components/EsHorizontalList.vue:32]: The design has the chevrons a bit closer to the circles. I doubt this design passes the tap-targets accessibility test as those circles and chevrons are pretty close together, but that part is out of scope - to better match the design without making the chevrons too close, maybe we bump this `mr-200` down to `mr-100`?

File: es-vue-base/src/lib-components/EsHorizontalList.vue [hunk 2]
```diff
@@ -46,7 +50,11 @@
             <es-button
                 v-if="arrows"
                 variant="link"
-                class="text-black d-none d-lg-block p-0 ml-200"
```
Comment (ericdouglaspratt) [inline es-vue-base/src/lib-components/EsHorizontalList.vue:49]: Similar here - maybe `ml-100`?

**Review by ericdouglaspratt — CHANGES_REQUESTED** [review]
Looks good - two small tweaks.

File: es-vue-base/src/lib-components/EsHorizontalList.vue (hunk 1 above)
Comment (nathanielwarner) [reply es-vue-base/src/lib-components/EsHorizontalList.vue:32]: Ah yes good catch!

**Review by ericdouglaspratt — APPROVED** [review]
Nice!
